### PR TITLE
🐛 Do not reload configs or files unnecessarily

### PR DESCRIPTION
**Pull request by fwkoch**
When consuming file warnings there was a bug that sometimes made them disappear. You see the log message that the warning occurred, but the store no longer has any record of the warning when processing is complete.

This was caused by `getRawFrontmatterFromFile` always re-loading files, even if they were already loaded, which in the process clears all previous warnings. Simply not re-loading the file if it is already loaded prevents this bug.

### CODE REVIEW

**Comment by fwkoch** [comment]
The scope of this PR has expanded to address the repeated config loading (and validation error messages) from #644

File: packages/myst-cli/src/store/reducers.ts [hunk 1]
```diff
@@ -39,18 +39,13 @@ export const config = createSlice({
...
     currentSitePath: string | undefined;
-    rawConfigs: Record<string, Record<string, any>>;
+    rawConfigs: Record<string, { raw: Record<string, any>; validated: Record<string, any> }>;
     projects: Record<string, ProjectConfig>;
     sites: Record<string, SiteConfig>;
     siteTemplateOptions: Record<string, SiteTemplateOptions>;
     filenames: Record<string, string>;
   },
   reducers: {
-    reload(state) {
```
Comment (fwkoch) [inline packages/myst-cli/src/store/reducers.ts:49]: Every time we `loadConfigs`, we compare the raw config to the cached raw config and, if it has changed, reload all the site/project configs.

This means we do not need to clear out the state like this; reloading will replace outdated configs.

File: packages/myst-cli/src/store/reducers.ts [hunk 2]
```diff
@@ -39,18 +39,13 @@ export const config = createSlice({
   } as {
     currentProjectPath: string | undefined;
     currentSitePath: string | undefined;
-    rawConfigs: Record<string, Record<string, any>>;
+    rawConfigs: Record<string, { raw: Record<string, any>; validated: Record<string, any> }>;
```
Comment (fwkoch) [inline packages/myst-cli/src/store/reducers.ts:42]: Previously, the "raw config" that we saved in the store was a partially validated/coerced version of the actual config on disk. That meant to see if the config on disk changed, we had to re-validate before comparing.

Now, we still save the validated config (for using everywhere that we want a validated config), but we also save the totally raw, unvalidated config which we can compare to the file and check for changes without re-running any validation code.

File: packages/myst-cli/src/frontmatter.ts [hunk 3]
```diff
@@ -83,7 +83,7 @@ export function prepareToWrite(frontmatter: { license?: Licenses }) {
 
 export async function getRawFrontmatterFromFile(session: ISession, file: string) {
   const cache = castSession(session);
-  await loadFile(session, file);
+  if (!cache.$mdast[file]) await loadFile(session, file);
```
Comment (fwkoch) [inline packages/myst-cli/src/frontmatter.ts:86]: This tiny change is all that was required to fix the bug addressed in the top PR comment 😆

File: packages/myst-cli/src/store/reducers.ts (hunk 2 above)
Comment (rowanc1) [reply packages/myst-cli/src/store/reducers.ts:42]: Nice, this will be helpful in the future as well!

**Comment by rowanc1** [comment]
Tested out, works well!

**Pull request merged**